__tests__/components/ProductList.js

```diff
@@ -0,0 +1,64 @@
+import sinon from 'sinon';
+import assert from 'assert';
+import React from 'react';
+import utils from 'react-addons-test-utils';
+import { ProductList } from '../../src/components/ProductList';
+
+describe('component ProductList', () => {
+    let props;
+    let productList;
+
+    function renderProductList() {
+        const renderer = utils.createRenderer();
+        renderer.render(<ProductList {...props} />);
+        const output = renderer.getRenderOutput();
+
+        productList = {
+            props,
+            output,
+            renderer,
+        };
+    }
+
+    beforeEach(() => {
+        props = {
+            products: [],
+            isFetching: false,
+            didInvalidate: false,
+            fetchProductsIfNeeded: sinon.spy(),
+        };
+
+        renderProductList();
+    });
+
+    it('should render correctly if no products', () => {
+        const { output } = productList;
+        const text = output.props.children;
+        assert.deepEqual(text.props.children, 'No product available.');
+    });
+
+    it('should render products correctly', () => {
+        props.products = [{
+            thumbnail: 'some url',
+            price: 'THE PRICE',
+            stock: 'THE STOCK',
+        }];
+        renderProductList();
```

Comment:
instead of using a (semi-) global `props`, it's be better to pass the custom props as parameters to the `renderProductList()` function. Not blocking for merge, but try avoiding globals or large scopes in the future, it's a code smell.